[17.0][MIG] barcodes_generator_abstract: Migration to 17.0

isaako34 opened the pull request:
Migrate the module barcodes_generator_abstract to V17.

This PR supersedes the [PR#593](https://github.com/OCA/stock-logistics-barcode/pull/593)

legalsylvain committed:
create new module 'barcodes_generator_abstract

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[MIG] barcodes_generator_abstract: Upgrade to v10
* Rename manifest
* Change openerp references to odoo
* Bump version
* Alphabetize imports
* pyBarcode is no longer maintained, switch to viivakoodi fork (https://bitbucket.org/whitie/python-barcode/issues/16/pypi-08-release-request#comment-33978213)

lasley committed:
[10.0][IMP] barcodes_generator_abstract: Add automated generator (#77)

Add an option to allow for automatic generation of barcode when a record of mixin model is created

lasley committed:
[10.0][FIX] barcodes_generator_abstract: Caching issue (#103)

* [FIX] barcodes_generator_abstract: Caching issue
* Make cache method return ids & move the recordset method out of cache to fix #93

phatnguyenuit committed:
[MIG] barcodes_generator_abstract: Migration to 11.0

legalsylvain committed:
[FIX] do not copy barcode_base during duplication

cvinh committed:
[MIG] barcodes_generator_abstract: Migration to 12.0

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

AndreVolksdorf committed:
Translated using Weblate (German)

Currently translated at 29.6% (8 of 27 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/de/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 14.8% (4 of 27 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/sl/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 46.4% (13 of 28 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/it/

alexis-via committed:
barcodes_generator_abstract: Fix warning

Improve form view of barcode rules
Improve README
Auto-add admin to the group

CarlosRoca13 committed:
[IMP] barcodes_generator_abstract: black, isort, prettier

CarlosRoca13 committed:
[MIG] barcodes_generator_abstract: Migration to v13.0

oca-travis committed:
[UPD] Update barcodes_generator_abstract.pot

Zar21 committed:
[IMP] barcodes_generator_abstract: black, isort, prettier

Zar21 committed:
[MIG] barcodes_generator_abstract: Migration to 14.0

oca-travis committed:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[UPD] README.rst

hugho-ad committed:
[FIX] barcodes_generator_abstract: Duplicate implicit target name: note

OCA-git-bot committed:
[UPD] README.rst

kevinkhao committed:
[FIX] remove redundant README fragment, fix dependencies in manifest

OCA-git-bot committed:
barcodes_generator_abstract 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (28 of 28 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-barcodes_generator_abstract/it/

AshishHirapara committed:
[IMP][14.0] barcodes_generator_abstract

updated images path on configuration readme file

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
barcodes_generator_abstract 14.0.1.0.2

legalsylvain committed:
[MIG] barcodes_generator_abstract from 14 to V16

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

Commit:
[UPD] Update barcodes_generator_abstract.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

FranciscoFactorLibre committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/es/

ilyasProgrammer committed:
[FIX] barcodes_generator_product: #516

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
barcodes_generator_abstract 16.0.1.0.2

legalsylvain committed:
[IMP] barcode_generator_abstract : UX : set the field barcode_rule.generate_model required if generate_type is defined

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : Do not display barcode_rule.generate_automate for non 'sequence' generation

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : Adds a message to specify that sequence creation can be automatic.

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : display barcode.rule new field on barcode.nomenclature form view

legalsylvain committed:
[REF] barcodes_generator_abstract : Refactor. Do not mix automatic generation and default value concepts:
- allow to have multiple automatic barcode.rule for a same model
- Simplify a lot the code base
- remove useless cache clearing

legalsylvain committed:
[REF] barcode_generate_abstract : Add more tests

legalsylvain committed:
[REF] Improve fields description

legalsylvain committed:
[REF] barcodes_generator_abstract : Use external_dependencies_override in setup file to correctly define python library

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
barcodes_generator_abstract 16.0.3.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

legalsylvain committed:
[FIX] barcodes_generator_abstract : align create algorithm with write algorightm

OCA-git-bot committed:
barcodes_generator_abstract 16.0.3.0.1

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/es/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.2% (17 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

RodrigoBM committed:
[FIX] barcodes_generator_abstract: get the real integer code for all coding types
if you use for example the gs1-128 coding, this coding cannot be saved in odoo directly.
>>> import barcode
>>> EAN = barcode.get_barcode_class('gs1_128')
>>> str(EAN('0000'))
'ñ0000'
We have to call the function get_fullcode which is a code that we can save in Odoo.
>>> import barcode
>>> EAN = barcode.get_barcode_class('gs1_128')
>>> str(EAN('0000').get_fullcode())
'0000'

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[FIX] barcodes_generator_*: Linter fixes

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

isaako34 commented:
@luisg123v Could you please review.

pedrobaeza commented:
/ocabot migration barcodes_generator_abstract

legalsylvain commented on requirements.txt:
```diff
@@ -1,2 +1,2 @@
 # generated from manifests external_dependencies
-barcode
+python-barcode
```
@sbidoul in the new v17 package design, what is the good way to map barcode to python-barcode ?

sbidoul replied:
Since v13 (or even 12, I don't remember), if you want https://pypi.org/project/python-barcode/, you need to declare `python-barcode` in the manifest `external-dependencies`. There is no mapping to do. The options that could be set in setup.py can still be set in pyproject.toml but I don't think that's ever needed in practice since v13.

legalsylvain replied:
Here is the context : 
1) if we write ```python-barcode``` we have the following error, when trying to install the module

```
  File "env-17.0/src/odoo/odoo/addons/base/models/ir_module.py", line 356, in check_external_dependencies
    raise UserError(msg)
odoo.exceptions.UserError: Unable to install module "barcodes_generator_abstract" because an external dependency is not met: Python library not installed: python-barcode

```
- https://github.com/odoo/odoo/blob/17.0/odoo/addons/base/models/ir_module.py#L351
- https://github.com/odoo/odoo/blob/17.0/odoo/modules/module.py#L465

2) If we write ```barcode```, we install a bad library so, we don't have required function. See : https://github.com/OCA/stock-logistics-barcode/pull/615/commits/1e225b7018213499caf2de2e34b6d9c9699ca599


> The options that could be set in setup.py can still be set in pyproject.toml but I don't think that's ever needed in practice.

The options was used for this module in [V12](https://github.com/OCA/stock-logistics-barcode/blob/12.0/setup/barcodes_generator_abstract/setup.py#L6), [V16](https://github.com/OCA/stock-logistics-barcode/blob/16.0/setup/barcodes_generator_abstract/setup.py#L6), etc...

See previous discussion : https://github.com/OCA/stock-logistics-barcode/pull/258

Do you know the right way to do it via pyproject.toml ?

sbidoul replied:
@legalsylvain I don't understand. This PR has python-barcode in external-dependencies, and the tests are green. All looks good, no?

legalsylvain replied:
oh. Indeed, I installed python-barcode locally and now all is good. In previous Odoo release, the replacement was mandatory. I don't understand why now all is OK. but it's great ! 

sorry for the noice. thanks for your answer. Have a nice week-end.

sbidoul replied:
Great. Note it was necessary until v12, but since then we can put the pypi name in external-dependencies.

luisg123v replied:
Yes. Actually, using the PyPI package is the recommended way ever since (see https://github.com/odoo/odoo/pull/25549), not the other way around. BTW, I just noticed that change was implemented by @sbidoul himself :)

luisg123v commented on barcodes_generator_abstract/views/view_barcode_rule.xml:
```diff
@@ -14,40 +14,28 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
                 <group
                     string="Barcode Generation"
-                    attrs="{'invisible': ['|', ('encoding', '=', 'any'), ('padding', '=', 0)]}"
+                    invisible="encoding == 'any' or padding == 0"
```
`or not padding`

luisg123v commented on test-requirements.txt:
```diff
@@ -0,0 +1 @@
+odoo_test_helper
```
Remove this commit and move this change to the migration commit. There should be only one commit addressing migration changes.

luisg123v commented on barcodes_generator_abstract/__manifest__.py:
```diff
@@ -21,5 +21,5 @@
     ],
     "demo": ["demo/res_users.xml"],
-    "external_dependencies": {"python": ["barcode"]},
+    "external_dependencies": {"python": ["python-barcode"]},
```
Check if this change works in v16. If so (most likely case), apply this in that version. If no, integrate into the migration commit.

isaako34 committed:
[IMP] barcodes_generator_abstract: set 'python-barcode' as external_dependencies

Since using the PyPI package has been the recommended way since v12.0
(see https://github.com/odoo/odoo/pull/25549), this commit sets
'python-barcode' as an external dependency and removes the
'odoo_addon' key from setup.py

OCA-git-bot committed:
[BOT] post-merge updates

isaako34 committed:
[IMP] : pre-commit auto fixes

isaako34 committed:
[MIG] barcodes_generator_abstract: Migration to 17.0

- Migrate to 17.0
- Add the package 'odoo_test_helper' to test-requirements.txt

isaako34 commented:
@luisg123v  ready for review

luisg123v reviewed (approved):
LGTM :+1: 

@legalsylvain what do you think?

legalsylvain reviewed (approved):
Diff review. LGTM. No test.

Thanks !

luisg123v commented:
@moylop260 could you review/merge, please?

moylop260 commented:
/ocabot merge nobump
